scripts/imgtool/image.py

```diff
@@ -164,20 +164,25 @@ def load(self, path):
 
         self.check()
 
-    def save(self, path):
+    def save(self, path, hex_addr=None):
         """Save an image from a given file"""
-        if self.pad:
-            self.pad_to(self.slot_size)
-
         ext = os.path.splitext(path)[1][1:].lower()
         if ext == INTEL_HEX_EXT:
             # input was in binary format, but HEX needs to know the base addr
-            if self.base_addr is None:
+            if self.base_addr is None and hex_addr is None:
                 raise Exception("Input file does not provide a base address")
```

Comment:
neither Input file nor parameter does provide a base address (please correct syntax as I'm not english native)

Reply:
Should have a better message now.